src/mpi/coll/allgatherv/allgatherv.c

```diff
@@ -299,17 +338,40 @@ int MPIR_Allgatherv(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
         
     if (comm_ptr->comm_kind == MPIR_COMM_KIND__INTRACOMM) {
         /* intracommunicator */
-        mpi_errno = MPIR_Allgatherv_intra(sendbuf, sendcount, sendtype,
+        switch (MPIR_Allgather_alg_intra_choice) {
+            case MPIR_ALLGATHERV_ALG_INTRA_BRUCKS:
+                mpi_errno = MPIR_Allgatherv_brucks(sendbuf, sendcount, sendtype,
                                           recvbuf, recvcounts, displs, recvtype,
                                           comm_ptr, errflag);
-        if (mpi_errno) MPIR_ERR_POP(mpi_errno);
+                break;
+             case MPIR_ALLGATHERV_ALG_INTRA_RECURSIVE_DOUBLING:
+                 mpi_errno = MPIR_Allgatherv_recursive_doubling(sendbuf, sendcount, sendtype,
+                                          recvbuf, recvcounts, displs, recvtype,
+                                          comm_ptr, errflag);
+                 break;
+             case MPIR_ALLGATHERV_ALG_INTRA_RING:
+                 mpi_errno = MPIR_Allgatherv_ring(sendbuf, sendcount, sendtype,
+                                          recvbuf, recvcounts, displs, recvtype,
+                                          comm_ptr, errflag);
+                 break;
+             case MPIR_ALLGATHERV_ALG_INTRA_AUTO:
+             default:
+                 mpi_errno = MPIR_Allgatherv_intra(sendbuf, sendcount, sendtype,
+                                          recvbuf, recvcounts, displs, recvtype,
+                                          comm_ptr, errflag);
+                break;
+        }
     } else {
-        /* intracommunicator */
+        switch (MPIR_Allgatherv_alg_inter_choice) {
+            default:
```

Comment:
I know I missed it above, but I'd like to see `case MPIR_ALLGATHERV_ALG_INTER_AUTO:` handled before the `default:` case here just in case we ever have other algorithms.

Obviously, that applies to all of the collectives too.